### PR TITLE
Bump Python version to 3.7

### PR DESCRIPTION
**Pull request by JeppeKlitgaard**
### Pull Request Checklist

- [X] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [X] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [X] I tested my changes.

### Description

I _think_ this covers most of a bump to Python 3.7.

### CODE REVIEW

**Review by Kwpolska — CHANGES_REQUESTED** [review]
3.7 is unacceptable (see discussion in #3532). 3.6 might be possible if there are other improvements that it would bring.

File: README.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ It has many features, but here are some of the nicer ones:
 * Syntax highlighting for almost any programming language or markup
 * Multilingual sites, `translated to 50 languages.`__
 * Doesn't reinvent wheels, leverages existing tools.
-* Python 3.4+ compatible.
```
Comment (Kwpolska) [inline README.rst:40]: Fixed (to say 3.5) in commit https://github.com/getnikola/nikola/commit/8cc31e0230a7209cc0bca558a9d085b2e179fe8c.

**Comment by JeppeKlitgaard** [comment]
Thanks for having a look at this @Kwpolska.

The reasoning behind 3.7 specifically was to get access to `__future__.annotations`, since I was hoping to expand on the type hinting in `Nikola`. Not knowing fully what the project's approach to legacy Python versions was, I thought it might be acceptable for the next major release, Nikola 9, to only support 3.7+, leaving 3.6 and 3.5 users 'stuck' on Nikola 8.

**Comment by Kwpolska** [comment]
We can change our minimum version version to 3.6 in a 8.x.0 release (3.5 was actually dropped in 8.0.4, a patch release), Nikola 9 isn’t really planned for now. Type annotations can be done without `from __future__ import annotations`, the required wrapping in strings isn’t that bad. We’d happily take a PR with 3.6+ type annotations and changing the minimum Python version at the same time.

**Comment by JeppeKlitgaard** [comment]
Abandoned for now

**Pull request closed without merging**